.github/workflows/kani.yml

```diff
@@ -245,3 +245,75 @@ jobs:
           if-no-files-found: error
           # Aggressively short retention: we don't really need these
           retention-days: 3
+
+  perf-benchcomp:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Kani (pull request HEAD)
+        uses: actions/checkout@v3
+        with:
+          path: ./pr
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Checkout Kani (pull request base)
+        uses: actions/checkout@v3
+        with:
+          path: ./base
+          ref: ${{ github.event.pull_request.base.sha }}
+
+      - name: Setup Kani Dependencies (pull request base)
+        uses: ./base/.github/actions/setup
+        with:
+          os: ubuntu-20.04
+          kani_dir: base
+
+      - name: Setup Kani Dependencies (pull request HEAD)
+        uses: ./pr/.github/actions/setup
+        with:
+          os: ubuntu-20.04
+          kani_dir: pr
+
+      - name: Build Kani (pull request HEAD)
+        run: pushd pr && cargo build-dev
+
+      - name: Build Kani (pull request base)
+        run: pushd base && cargo build-dev
+
+      - name: Write benchcomp config file
+        run: |
+          cat >benchcomp.yaml <<EOM
+          variants:
+            kani_pr:
+              config:
+                directory: pr
+                command_line: PATH=$(realpath pr/target/kani/bin):$PATH scripts/kani-perf.sh
+                env:
+                  RUST_TEST_THREADS: "1"
+            kani_base:
+              config:
+                directory: base
+                command_line: PATH=$(realpath base/target/kani/bin):$PATH scripts/kani-perf.sh
+                env:
+                  RUST_TEST_THREADS: "1"
+          run:
+            suites:
+              kani_perf:
+                parser:
+                  module: kani_perf
+                variants: [kani_base, kani_pr]
+          visualize:
+            - type: error_on_regression
+              variant_pairs: [[kani_base, kani_pr]]
+              checks:
+                - metric: success
+                  test: "lambda old, new: False if not old else not new"
+                - metric: solver_runtime
+                  test: "lambda old, new: False if new < 1 else new / old > 1.1"
+                - metric: symex_runtime
+                  test: "lambda old, new: False if new < 1 else new / old > 1.1"
+                - metric: verification_time
+                  test: "lambda old, new: False if new < 1 else new / old > 1.1"
```

Comment:
This may frequently block on benchmarks which take a small amount of time (e.g., less than 5s). Maybe we should define different scales, for example:
 * Less than 1s: Block on >3s
 * Less than 30s: Block on >1min
 * Anything else: Block on >1.1x
Of course, I haven't experimented on these particular values, so more scales or tweaks might be needed.